### ISSUE TITLE
New field type for modelling data relations

### ISSUE THREAD

**Issue by johannesjh** [lektor/lektor#452]
Do Lektor's models support relations between data models? I found the documentation about [lektor field types](https://www.getlektor.com/docs/api/db/types/), but there seems to be no way to establish links between models. The "flow" field type seems to somewhat similar, but it can only link to flowblocks, not to other full-fledged models, as it seems.

Suggestion: Would a new field type make sense, that allows to:

- 1:n relationships: Link to another model (either restricted to a specific model type, or open for any type of model). Lektor's admin GUI could display radiobuttons or an autocompleting dropdown list for choosing the other model.
- m:n relationshipts: Link to multiple other model instances . Lektor's admin GUI could display checkboxes or an autocompleting multi-select control as editor in Lektor's admin GUI.

Disclaimer, and background of this question: I am new to Lektor. Its data modelling features caught my attention because I am interested in systems for working with information, including documenting stuff, visualizing it, evolving the schema to add or restructure information, linking between entities, and visualizing it again to develop a better understanding. Given this background, I am excited to learn more about Lektor's "data modelling" features.

**Comment by barryrowlingson** [comment]
+1

 I have a site that has a number of people and a number of projects. There's an n:m relationship between people and projects. In my current CMS (Hyde), each project has a list of people-ids in the front-matter YAML. The template for the project gets this list and creates a sidebar box of the people working on this project.

I think you could do this in lektor by having a strings field in the project model and looking up the people that way, but a proper object-relation field would prevent typos and could be really nice.

**Comment by johannesjh** [comment]
> I think you could do this in lektor by having a strings field in the project model and looking up the people that way, but a proper object-relation field would prevent typos and could be really nice.

yes, managing such relationships is simple at first. But it quickly requires special functionality, additional logic, and UIs:

- Suitable user interfaces, e.g., for displaying and choosing related entities. 
- Logic for data maintencance, e.g., deleting dependent entities when their parent is deleted (in case it is a hard dependency), and warnings about broken relationships. 
- Configuration options because of performance considerations, e.g., whether related entities should be loaded in an eager or lazy way.
- Serialization/deserialization issues with cyclic relationships (not sure if such problems would concern lektor).

...hence the **question to lektor's creators and maintainers: are data relationships part of lektor's roadmap?** (if not part of its current featureset?)

**Comment by nixjdm** [comment]
This is not part of the current feature set. The closest things are [flowblocks](https://www.getlektor.com/docs/content/flow/) and [categories](https://www.getlektor.com/docs/guides/categories/), or crafting your own field lookups. I think I'd like to see something like this added too, but we've got some other goals we're working on that need to happen first. I'll mark this as an enhancement and leave it open, but we probably won't get to this one soon.

**Comment by johannesjh** [comment]
ok thank you, I appreciate your time and consideration!

**Comment by xlotlu** [comment]
I personally find this pretty important for a real CMS. It should be handled along with #625.
